### DISCUSSION TITLE
Console - Terminal Type = IBM-3278-2-E

### DISCUSSION THREAD

**Discussion by Claudemar** [PerryWerneck/pw3270#62] [General]
Bom dia Perry Werneck, uma duvida?

Usamos o TN3270 Plus aqui só para a console, dentro dele tem uma configuração:

Telnet Terminal Type = IBM-3278-2-E

No PW3270 existe algo para habilitar tipo de configuração para a gente conseguir conectar a console?

Pesquisei no help do PW3270, algum desses parametros altera essa opção?

``Usage:
pw3270.exe [OPTION…] - 3270 Emulator for Gtk

Help Options:
-?, --help Show help options

Application Options:
-a, --appname=pw3270 Application name
-d, --datadir Path to application data files
-s, --session=pw3270 Session name
-h, --host Host to connect
-c, --colors=16 Set reported colors (8/16)
-t, --systype=S390 Host system type
-S, --toggleset Set toggles ON
-R, --togglereset Set toggles OFF
-C, --charset Set host charset
-m, --remap Remap charset from xml file
-M, --model The model of 3270 display to be emulated
-D, --autodisconnect Minutes for auto-disconnect
-P, --pluginpath Path for plugin files
-T, --tracefile Set trace filename
-L, --log Log to file``

**Comment by PerryWerneck** [comment]
3278 pelo que eu me lembro é terminal monocromático. Você chegou a usar --colors=1 como opção nessa versão antiga? 

Ou setar o tipo de terminal para 'monocromático' no diálogo de configuração da sessão nas versões mais recentes.

Fiz um teste aqui na versão mais recente habilitando terminal monocromático e ele negociou o formato 3278-2-E, aparece a linha abaixo no trace:

`
RCVD SB TN3270E DEVICE-TYPE IS IBM-3278-2-E CONNECT
`

**Reply by Claudemar** [reply]
Vou fazer um teste no computador onde roda o emulador da console, valeu pela ajuda Perry

**Reply by SergioCoutinhoBR** [reply]
Olá,

Estou com o mesmo problema, não consigo conectar a console.
Alguma dica do que fazer?

**Reply by PerryWerneck** [reply]
> Estou com o mesmo problema, não consigo conectar a console. Alguma dica do que fazer?

Mudou o tipo de terminal para monocromático como sugerido?

**Reply by SergioCoutinhoBR** [reply]
Sim, mas não conecta.
Estou em um servidor LINUX RedHat 8.5 utilizando o emulador HERCULES, executando tudo no mesmo servidor.

[root@srvlnxwla001 hyperion]# ./hercules
HHC01413I Hercules version 4.7.0.11119-SDL-gf7d2360a
HHC01414I (C) Copyright 1999-2024 by Roger Bowler, Jan Jaeger, and others
HHC01417I ** The SDL 4.x Hyperion version of Hercules **
HHC01415I Build date: Dec 16 2024 at 17:54:20
HHC01417I Built with: GCC 8.5.0 20210514 (Red Hat 8.5.0-22.0.1)
HHC01417I Build type: GNU/Linux x86_64 host architecture build
HHC01417I Modes: S/370 ESA/390 z/Arch
HHC01417I Max CPU Engines: 128

**Reply by SergioCoutinhoBR** [reply]
Aversão que estou utilizando é a 5.5-24.10.17.
E chamo o terminal desta forma:

$ flatpak run br.app.pw3270.terminal

**Reply by SergioCoutinhoBR** [reply]
Testei, da mesma forma, ou seja, tudo no mesmo servidor, na plataforma Windows 11 e tudo funciona perfeitamente.

**Reply by PerryWerneck** [reply]
E qual erro aparece quando tenta conectar?

**Reply by SergioCoutinhoBR** [reply]
Aí que está, não dá erro, mas não conecta como console.
Sigo os seguintes passos:
1. Inicio o Hercules;
2. Abro o terminal pw3270 (a tela principal aparece com todas as informações);
3.  e as mensagens abaixo aparecem na tela do Hercules:

HHC02915I client 0 COMM: Connection received
HHC02914I 0:001F COMM: client 0 negotiations complete; ttype = 'IBM-3278-2-E'
HHC01018I 0:001F COMM: client 172.25.11.237 devtype 3270: connected


4. Executo um ipl no ambiente do Hercules;
5. E nada acontece......o terminal aberto não herda a console.

**Reply by SergioCoutinhoBR** [reply]
![Tela pw2370](https://github.com/user-attachments/assets/ffb9c1ee-1f7b-4a2e-b245-c4e635afbc93)

**Reply by SergioCoutinhoBR** [reply]
Não sai dessa tela.

**Reply by SergioCoutinhoBR** [reply]
E qualquer tecla que eu aperte aparece um X amarelo ao lado do 4A+(quadrado verde)

**Reply by PerryWerneck** [reply]
Uai, mas está conectado normalmente! O X amarelo indica que o host mandou uma mensagem de terminal bloqueado. O problema me parece ser no hercules e não no pw3270.

**Reply by SergioCoutinhoBR** [reply]
Vou pedir ajuda aos universitários de mainframe aqui......., mas volto informando a solução, se funcionar. 
Obrigado pela ajuda.

**Reply by Claudemar** [reply]
> HHC01414I (C) Copyright 1999-2024 by Roger Bowler, Jan Jaeger, and others

Por causa do serviço tenho dois emuladores de telnet, o PW3270 e o X3270, teste o X3270 para ver se o mesmo problema ocorre

**Reply by Claudemar** [reply]
Estou usando hoje o PW3270 nos dois, Hercules e para o serviço, estou acessando de boa os dois
